### ISSUE TITLE
Implement named access on Window (named properties object, global scope polluter)

### ISSUE THREAD

**Issue by Ms2ger** [servo/servo#8984]
- Specs:
  - https://heycam.github.io/webidl/#named-properties-object
  - https://html.spec.whatwg.org/multipage/#named-access-on-the-window-object
- Tests:
  - html/browsers/the-window-object/test_window-named-properties.html
  - html/browsers/the-window-object/test_window-prototype-chain.html
- Gecko code:
  - https://mxr.mozilla.org/mozilla-central/source/dom/base/WindowNamedPropertiesHandler.h
  - https://mxr.mozilla.org/mozilla-central/source/dom/base/WindowNamedPropertiesHandler.cpp

**Comment by nox** [comment]
Blocked by #8954.

**Comment by delan** [comment]
(closing in favour of #27949)
